### PR TITLE
8329340: Remove unused libawt code

### PR DESCRIPTION
**Pull request by djelinski**
Please review this PR that removes unused functions, variables, and WM_AWT window messages.

The unused code was detected by automated analysis (MSVC compiler and linker, CodeQL analyzer). I manually verified every symbol before removing. Mach5 client libs testing clean.

Some WM_AWT messages have different IDs after this change. The IDs have changed a few times before, so I think this shouldn't be a problem.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8329340](https://bugs.openjdk.org/browse/JDK-8329340): Remove unused libawt code (**Bug** - P4)
 * [JDK-8315693](https://bugs.openjdk.org/browse/JDK-8315693): Remove WM_AWT_SET_SCROLL_INFO message (**Bug** - P5)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [cbf7bbd3](https://git.openjdk.org/jdk/pull/18553/files/cbf7bbd31d4fa6f5b159d2352042a8e09ad78d5b)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18553/head:pull/18553` \
`$ git checkout pull/18553`

Update a local copy of the PR: \
`$ git checkout pull/18553` \
`$ git pull https://git.openjdk.org/jdk.git pull/18553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18553`

View PR using the GUI difftool: \
`$ git pr show -t 18553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18553.diff">https://git.openjdk.org/jdk/pull/18553.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18553#issuecomment-2027740213)

### CODE REVIEW

**Comment by prrace** [comment]
About the WM_AWT messages, how did these tools you mention figure out no message of that name / ID is ever created ?
I presume that's what you must have done to decide it is OK to remove the handling of these messages.

**Comment by djelinski** [comment]
> About the WM_AWT messages, how did these tools you mention figure out no message of that name / ID is ever created ?
I presume that's what you must have done to decide it is OK to remove the handling of these messages.

Well, that part was actually me. The linker reported that `AwtObject::WinThreadExec` was unused; while removing it, I found that `WM_AWT_EXECUTE_SYNC` is no longer needed, so I removed it as well. Then I figured that there may be other left over WM_AWT messages, so I checked the entire enum in awtmsg.h. I removed every enum constant that was never used outside of the WIN_MSG macro or a case label.

I checked the history for some of the messages; WM_AWT_FORWARD_BYTE was last used before e4f977c0a1663dd71aa19019d55b07d87ad667e2, and the last use of WM_AWT_SHOWCURSOR / WM_AWT_HIDECURSOR was removed in e4c9db984fe1a3a9d06fb60e333e92df3b165454. WM_AWT_SET_SCROLL_INFO was not used anywhere in git history. The other removed enums aren't used anywhere.

**Review by prrace — APPROVED** [review]
Good to push this early in the week in case the testing was some how not enough.

**Comment by djelinski** [comment]
Thanks for the review! Should I wait for a second one?

Also, what's your definition of "early in the week"? Should I wait for Monday, or does anything other than Friday qualify?

File: src/java.desktop/windows/native/libawt/windows/awtmsg.h [hunk 1]
```diff
@@ -209,23 +209,16 @@ enum {
...
     WM_AWT_DESTROYCONTEXT,
     WM_AWT_ASSOCIATECONTEXT,
     WM_AWT_GET_DEFAULT_IME_HANDLER,
     WM_AWT_HANDLE_NATIVE_IME_EVENT,
-    WM_AWT_PRE_KEYDOWN,
-    WM_AWT_PRE_KEYUP,
-    WM_AWT_PRE_SYSKEYDOWN,
-    WM_AWT_PRE_SYSKEYUP,
 
     /* deleted DND mesg's */
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awtmsg.h:224]: I guess this comment can safely be removed. What does it add? There had been some messages that were removed a long time ago.

**Review by aivanov-jdk — APPROVED** [review]
Looks good to me.

**Comment by aivanov-jdk** [comment]
You should add [JDK-8315693](https://bugs.openjdk.org/browse/JDK-8315693): _Remove WM_AWT_SET_SCROLL_INFO message_ to the list of resolved issues.

**Comment by aivanov-jdk** [comment]
You should also update the copyright years where applicable.

File: src/java.desktop/windows/native/libawt/windows/awt_Debug.cpp [hunk 2]
```diff
@@ -70,60 +70,6 @@ void operator delete(void *ptr) throw() {
...
-    DASSERT(msg != NULL);
-
-    ::GetClipBox(hdc, &r);
-    DTrace_PrintImpl("%s: hdc=%x, %d %d %d %d\n", msg, hdc, r.left, r.top, r.right, r.bottom);
-}
-
-/*
- * DTRACE print callback to dump window's update region bounding rectangle
- */
-void DumpUpdateRectangle(const char * file, int line, int argc, const char * fmt, va_list arglist) {
```
[the remark this answers is not in the document]
Comment (djelinski) [reply src/java.desktop/windows/native/libawt/windows/awt_Debug.cpp:111]: they were `if defined(DEBUG)` already; I can keep them if you think it makes sense.

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/windows/native/libawt/windows/awt_Debug.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/java.desktop/windows/native/libawt/windows/awt_Debug.cpp:111]: I don't have any strong preference. These may be useful in rare cases where painting doesn't work as expected…

**Pull request closed without merging**